keps/11823-effective-resource-groups/README.md

```
@@ -0,0 +1,278 @@
+The major risks here are allowing *spec.ResourceGroups* to be empty and allowing a period of time where *status.EffectiveResourceGroups* might be empty as well.
+
+Mitigating this requires an expansive sweep of existing logic, ensuring the code is made secure against:
+* referencing an empty *spec.ResourceGroups* or *status.EffectiveResourceGroups* properties,
+* a misalignment/delay of *status.EffectiveResourceGroups* being synced to *spec.ResourceGroups*.
+
+## Design Details
+
+*status.EffectiveResourceGroups* will be managed by either the Core ClusterQueue Controller or the MultiKueue ClusterQueue Controller (mutually exclusive):
+1. *status.EffectiveResourceGroups* will be managed by the MultiKueue Controller when (AND):
+	* The MultiKueue ClusterQueue Controller is active:
+    * features.MultiKueue is enabled
```

markdownlint-cli2 (0.22.1)

[warning] Inconsistent indentation for list items at the same level
Expected: 1; Actual: 4

(MD005, list-indent)